File: 7.cs
ConsoleColor foreground = Console.ForegroundColor;
Console.ForegroundColor = ConsoleColor.Green;

int num_getter()
{
    int num;
    bool flag1 = false, val;
   
    do
    {
        Console.Write("\n\nENTER A NUMBER FROM 1 TO 12 (0 TO EXIT): ");
        val = int.TryParse(Console.ReadLine(), out num);
        if (num < 0 || num > 12) { flag1 = true; }
        else if (num == 0 && val) 
        {
            Console.WriteLine("\n\nGOODBYE! :)\n\n");
            System.Environment.Exit(0);
        }
        else {  flag1 = false; }
    
    } while (flag1 == true || val == false);
    
    return num;
}

int global_num, i = 0, j;
string format, format2;
Console.WriteLine("*********************************");
Console.WriteLine("      MULTIPLICATION TABLES      ");
Console.WriteLine("*********************************\n\n");
global_num = num_getter();
Console.WriteLine("\n\n -------------------");
while (i != global_num)
{
    j = i * global_num;
    format = String.Format("{0, -2} {1, -2} {2, -2} {3, -2}", i, " X ", global_num, " = ");
    format2 = String.Format("{0, -3} {1, -3}", j, " |");
    Console.WriteLine("| " + format + format2);
    i++;
}
j = i * global_num;
format = String.Format("{0, -2} {1, -2} {2, -2} {3, -2}", i, " X ", global_num, " = ");
format2 = String.Format("{0, -3} {1, -3}", j, " |");
Console.WriteLine("| " + format + format2);
Console.WriteLine(" -------------------");

return 0;

File: 7.java
import java.text.NumberFormat;
import java.text.ParsePosition;
import java.util.Scanner;

public class Main {
	
	public static final String ANSI_GREEN = "\u001B[32m";
	public static final String ANSI_RESET = "\u001B[0m";
	public static final String ANSI_BLACK_BG = "\u001B[40m";
	static Scanner readLine = new Scanner(System.in);
	
	
	public static int numGetter() {
		
		String data; double num = 0; boolean flag1 = true;
		
		do {
			System.out.print("\n\nENTER A NUMBER FROM 1 TO 12 (0 TO EXIT): ");
			data = readLine.nextLine();
			data = data.trim();
			if (numCheck(data)) {
				num = Double.parseDouble(data);
				flag1 = false;
				if (num == 0) {
					System.out.println("\n\nGOODBYE! :)\n");
					System.exit(0);
				}
			}
		} while (flag1 == true);
		
		return (int)num;
	}
	
	public static boolean numCheck(String str) {
		
		ParsePosition pos = new ParsePosition(0);
		NumberFormat.getInstance().parse(str, pos);
		return str.length() == pos.getIndex();
	}

	public static void main(String[] args) {
		
		System.out.println("*********************************");
		System.out.println("      MULTIPLICATION TABLES      ");
		System.out.println("*********************************\n");
		
		int number = numGetter();
		
		System.out.println("\n\n -------------------");
		
		int i = 0, j;
		String strTable, strTable2;
		while (i != number) {
			j = i * number;
			strTable = String.format("%-2d %-2s %-2d %-2s", i, " X ", number, " = ");
			strTable2 = String.format("%-3d %-3s", j, " |");
			System.out.println("| " + strTable + strTable2);
			i++;
		}
		j = i * number;
		strTable = String.format("%-2d %-2s %-2d %-2s", i, " X ", number, " = ");
		strTable2 = String.format("%-3d %-3s", j, " |");
		System.out.println("| " + strTable + strTable2);
		System.out.println(" -------------------");
	}
}
